### PR TITLE
Update to Node.js v18 LTS

### PR DESCRIPTION
**Pull request by gjvoosten**
Update to current Node.js LTS release (v18).
This means our wdio tests needed to be rewritten from sync to async.
Updated wdio/webdriverio to v8.

Closes [AB#339](https://dev.azure.com/ncia-anet/2aa083a5-af3d-44e1-8c7b-6e9e6b124d91/_workitems/edit/339)

#### User changes
- None

#### Super User changes
- None

#### Admin changes
- None

#### System admin changes
- [ ] anet.yml or anet-dictionary.yml needs change
- [ ] db needs migration
- [ ] documentation has changed
- [ ] graphql schema has changed

### Checklist
  - [x] Described the user behavior in PR body
  - [x] Referenced/updated all related issues
  - [x] commits follow a `repo#issue: Title` title format and [these 7 rules](https://chris.beams.io/posts/git-commit/)
  - [x] commits have a [clean history](https://epage.github.io/dev/commits/), otherwise PR may be squash-merged
  - [ ] Added and/or updated unit tests
  - [x] Added and/or updated e2e tests
  - [ ] Added and/or updated data migrations
  - [ ] Updated documentation
  - [x] Resolved all build errors and warnings
  - [ ] Opened debt issues for anything not resolved here

### CODE REVIEW

File: client/tests/webdriver/baseSpecs/advancedSearch.spec.js [hunk 1]
```diff
@@ -30,89 +30,105 @@ const PERSON_INDEX = 1
...
+      ).to.match(
+        new RegExp(ANET_OBJECT_TYPES[await getObjectType(i)].sampleFilter)
       )
-    })
+    }
   })
 })
 
-function getObjectType(index) {
+async function getObjectType(index) {
```
Comment (midmarch) [inline client/tests/webdriver/baseSpecs/advancedSearch.spec.js:132]: Does not look async to me

File: client/tests/webdriver/pages/createNewPosition.page.js [hunk 2]
```diff
@@ -2,95 +2,100 @@ import Page from "./page"
 
 const PAGE_URL = "/positions/new"
 class CreatePosition extends Page {
-  get form() {
+  async getForm() {
```
Comment (midmarch) [inline client/tests/webdriver/pages/createNewPosition.page.js:5]: not async

File: client/tests/webdriver/pages/createNewPosition.page.js [hunk 3]
```diff
@@ -2,95 +2,100 @@ import Page from "./page"
...
     return browser.$("form")
   }
 
-  get positionNameInput() {
+  async getPositionNameInput() {
```
Comment (midmarch) [inline client/tests/webdriver/pages/createNewPosition.page.js:9]: not async

File: client/tests/webdriver/pages/createNewPosition.page.js [hunk 4]
```diff
@@ -2,95 +2,100 @@ import Page from "./page"
...
     return browser.$("#name")
   }
 
-  get duplicatesButton() {
+  async getDuplicatesButton() {
```
Comment (midmarch) [inline client/tests/webdriver/pages/createNewPosition.page.js:13]: not async

File: client/tests/webdriver/pages/createNewPosition.page.js [hunk 5]
```diff
@@ -2,95 +2,100 @@ import Page from "./page"
...
     return browser.$('//button[text()="Possible Duplicates"]')
   }
 
-  get modalContent() {
+  async getModalContent() {
```
Comment (midmarch) [inline client/tests/webdriver/pages/createNewPosition.page.js:17]: not async

**Review by midmarch — CHANGES_REQUESTED** [review]
What a massive change!

Looks good, there are just a bunch of functions which may do return a promise, but they are async.

**Comment by gjvoosten** [comment]
> What a massive change!
> 
> Looks good, there are just a bunch of functions which may do return a promise, but they are async.

These were all added by the codemod, so let's decide to accept them as-is (instead of having to go through each and every one of them to see if they can be removed or not).

**Pull request merged**